internal/appconfig/serde.go

```diff
@@ -73,7 +78,42 @@ func (c *Config) WriteTo(w io.Writer, format string) (int64, error) {
 		}
 	}
 
-	return bytes.NewBuffer(b).WriteTo(w)
+	if format == "toml" {
+		return bytes.NewBuffer(b).WriteTo(w)
+	} else {
+		// pretty print by intelligently adding newlines
+		scanner := bufio.NewScanner(bytes.NewReader(b))
+		section := true
+		total := 0
+		for scanner.Scan() {
+			text := scanner.Text()
+
+			if startObjectOrArray.MatchString(text) {
+				if !section {
+					count, err := w.Write([]byte("\n"))
+					if err != nil {
+						return 0, err
+					}
+					total += count
+					section = true
+				}
+			} else {
+				section = false
+			}
+
+			count, err := w.Write([]byte(text + "\n"))
+			if err != nil {
+				return 0, err
+			}
+			total += count
+		}
+
+		if err := scanner.Err(); err != nil {
+			return 0, err
+		}
```

Comment:
I'd put all this stuff in a separate private function, name it accordingly and even add a simple test case so in case we found bugs in the regex for some edge case, a future ourselves could fix them with certainty of not breaking past fixes.